internal_packages/keybase/lib/pgp-key-store.cjsx

```diff
@@ -0,0 +1,391 @@
+NylasStore = require 'nylas-store'
+{DraftStore, MessageBodyProcessor, RegExpUtils} = require 'nylas-exports'
+{remote} = require 'electron'
+kb = require './keybase'
+pgp = require 'kbpgp'
+_ = require 'underscore'
+path = require 'path'
+fs = require 'fs'
+
+class PGPKeyStore extends NylasStore
+
+  constructor: ->
+    super()
+
+    @_pubKeys = {}
+    @_privKeys = []
+
+    @_msgCache = []
+    @_msgStatus = []
+
+    @_pubWatcher = null
+    @_privWatcher = null
+
+    @_keyDir = path.join(NylasEnv.getConfigDirPath(), 'keys')
+    @_pubKeyDir = path.join(@_keyDir, 'public')
+    @_privKeyDir = path.join(@_keyDir, 'private')
+
+    # Create the key storage file system if it doesn't already exist
+    fs.access(@_keyDir, fs.R_OK | fs.W_OK, (err) =>
+      if err
+        fs.mkdir(@_keyDir, (err) =>
+          if err
+            console.warn err
+          else
+            fs.mkdir(@_pubKeyDir, (err) =>
+              if err
+                console.warn err
+              else
+                fs.mkdir(@_privKeyDir, (err) =>
+                  if err
+                    console.warn err
+                  else
+                    @watch())))
+      else
+        fs.access(@_pubKeyDir, fs.R_OK | fs.W_OK, (err) =>
+          if err
+            fs.mkdir(@_pubKeyDir, (err) =>
+              if err
+                console.warn err))
+        fs.access(@_privKeyDir, fs.R_OK | fs.W_OK, (err) =>
+          if err
+            fs.mkdir(@_privKeyDir, (err) =>
+              if err
+                console.warn err))
+        @_populate(isPub = true)
```

Comment:
I think the way @logandavis wrote this makes sense - it seems to me like it's just a way to pass a boolean while also documenting what it's for instead of just having `somefunc(true)` in which case who knows what that true does.

Correct me if I'm wrong, Logan, but it makes sense to me.